### ISSUE TITLE
[Go] Parent classes and std::string break compilation

### ISSUE THREAD

**Issue by steeve** [swig/swig#339]
Hi,

Taking the following snippet:

```cpp
%module test

%include <std_string.i>

struct parent
{
    std::string foo;
};

struct child: parent
{
    std::string bar;
};
```

The generated code for `child->foo` will generate `foo` as `_gostring_*` instead of `_gostring_`, resulting in a break of compilation when reaching `swigCopyString` in the typemap for `std::string`.

Note that `parent->foo` and `child->bar` are properly generated as `_gostring_`.

**Comment by steeve** [comment]
Did some tracing.

For `child.foo`:
`gccFunctionWrapper` is called with `result=p.std::string`

For `child.bar`:
`gccFunctionWrapper` is called with `result=r.q(const).std::string`

**Comment by steeve** [comment]
The funny thing is that the node has the following properties:
```
Hash(0x10326ee40) {
  'wrap:action' : result = (std::string *) & ((arg1)->foo);,
  'sym:symtab' : Hash(0x10326ecc0) {......},
  'goBaseVariable:wrap:action' : result = (std::string *) & ((arg1)->foo);,
  'goBaseVariable:type' : std::string,
  'nextSibling' : Hash(0x103281a80) {.....},
  'wrap:name' : _wrap_parent_foo_get_test_74e8060a949a17b4,
  'kind' : variable,
  'go:type' : r.q(const).std::string,
  'type' : r.q(const).std::string,
  'access' : public,
  'ismember' : 1,
  'tmap:goout' :  swig_r_1 = swigCopyString(swig_r) ,
  'sym:overname' : __SWIG_0,
  'goBaseVariable:view' : variableHandler,
  'parms' : Hash(0x10328a4a0) {.....},
  'sym:name' : foo,
  'parentNode' : Hash(0x10326ebe0) {..........................},
  'nodeType' : cdecl,
  'feature:except' : 0,
  'view' : goBaseVariable,
  'name' : foo,
  'wrap:parms' : Hash(0x103282240) {................},
  'decl' : ,
  'tmap:goout:fragment' : CopyString,
}
```

Looking at the `type` key, everything looks normal.

**Comment by steeve** [comment]
I've traced it https://github.com/swig/swig/blob/master/Source/Modules/go.cxx#L2512

`std::string` being a class, the call is successful, and thus a pointer is added, breaking everything.

**Comment by steeve** [comment]
Changed with this hack, it works properly:
```
    if (Strcmp(Getattr(var, "tmap:gotype"), "string") != 0 && SwigType_isclass(vt)) {
      SwigType_add_pointer(vt);
    }
```

**Comment by steeve** [comment]
still broken in e4d02d20ad8900cc1c1df4dd3d8dbdcaa578b53a

@ianlancetaylor ?

**Comment by steeve** [comment]
any news on this? go 1.5 being real soon

**Comment by ianlancetaylor** [comment]
Fixed now.  Thanks for reporting it.

Note that this is only a SWIG fix--the timing of the Go 1.5 release is irrelevant to this.
